src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -18,7 +17,7 @@ internal class SNIMarsManager
         /// <summary>
         /// Constructor
         /// </summary>
-        public SNIMarsManager()
+        private SNIMarsManager()
```

Comment:
Not sure why it was written as a singleton. But it makes sense to me because we're actually tracking some state in the MarsManager, it's just that there only needs to be one instance tracking that state. We have lots of other Singleton types all over the SqlClient code too.